src/v2/api/services/PermissionService.ts

```diff
@@ -22,22 +22,21 @@ export class PermissionService {
     return this.checkPermission(roles, permission);
   }
 
-  hasRolePermission (grants: Grant[], permission: string) {
-    for (const grant of grants) {
-      const hasPermission = this.hasGrantPermission(permission, grant.permission);
-      if (hasPermission) return grant.set;
-    }
+  hasRolePermission (grants: Grant[], permission: string): boolean | undefined {
```

Comment:
The Law of Demeter was broken. Use Permissions right away instead of Grants, the rest of the entity is unnecessary here.

Reply:
Yes, it's a bit complicated logic, so i have changed the implementation of PermissionService at all :3